src/app/shared/spaces.service.ts

```diff
@@ -10,15 +10,15 @@ export class SpacesService implements Spaces {
   static readonly RECENT_SPACE_LENGTH = 8;
 
   private _current: Observable<Space>;
-  private _recent: ReplaySubject<Space[]> = new ReplaySubject<Space[]>(1);
-  private recentChanged: boolean = false;
+  private _recent: BehaviorSubject<Space[]>;
 
   constructor(
     private contexts: Contexts,
     private spaceService: SpaceService,
     private broadcaster: Broadcaster,
     private profileService: ProfileService
   ) {
+    this._recent = new BehaviorSubject<Space[]>([] as Space[]);
```

Comment:
Is the `as Space[]` necessary? Seems like `[]` alone should be sufficient for the compiler to infer the type...

Reply:
Yeah it should probably be enough, just figured I would throw them in wherever I could.

Reply:
I don't think throwing in type casts wherever they fit is a good habit, frankly...